discovery/file/file_test.go

```diff
@@ -59,30 +61,45 @@ func testFileSD(t *testing.T, prefix, ext string, expect bool) {
 		t.Fatalf("Unexpected target groups in file discovery: %s", tgs)
 	}
 
-	newf, err := os.Create("fixtures/_test_" + prefix + ext)
+	// To avoid empty group struct sent from the discovery caused by invalid fsnotify updates,
+	// drain the channel until we are ready with the test files.
+	filesReady := make(chan struct{})
+	go func() {
+	Loop:
+		for {
+			select {
+			case <-ch:
+			case <-filesReady:
+				break Loop
```

Comment:
Can't this just be `return`, since it's leaving the function afterward anyway?

Reply:
yeah.
it is failing again with travis so need to find another way.